pysidt/sidt.py

```diff
@@ -657,25 +657,19 @@ def generate_tree(
                 for child in children_to_remove:
                     node.children.remove(child)
 
-            self.fit_tree(data=None, check_data=False, alpha=alpha, fit_rule=True, estimate_full_uncertainty=True)
+            self.fit_tree(data=None, check_data=False, alpha=alpha)
 
-        else:
-
-            self.fit_tree(data=None, check_data=False, alpha=alpha, fit_rule=False, estimate_full_uncertainty=True)
-
-    def fit_tree(self, data=None, check_data=True, alpha=0.1, fit_rule=True, estimate_full_uncertainty=False):
+    def fit_tree(self, data=None, check_data=True, alpha=0.1):
         """
         fit rule for each node
         """
         if data:
             self.setup_data(data, check_data=check_data)
             self.descend_training_from_top(only_specific_match=True)
 
-        if fit_rule:
-            self.fit_rule(alpha=alpha)
+        self.fit_rule(alpha=alpha)
         
-        if estimate_full_uncertainty:
-            self.estimate_full_uncertainty()
+        self.estimate_full_uncertainty()
```

Comment:
Shouldn't this be just "estimate_uncertainty" now since there isn't a partial vs full anymore?